_projects/nextgen_action_system.md

```diff
@@ -0,0 +1,43 @@
+---
+name: "Nextgen action system"
+desc: "Actions in coala should be improved to allow for more dynamic actions"
+requirements:
+ - "The applicant must understand how results and actions work in coala."
+ - "The proposal needs to contain a clear concept."
+difficulty: "medium"
+mentors:
+ - Makman2
+ - jayvdb
+initiatives:
+ - GSoC
+tags:
+ - core
+---
+
+Currently, `Result` objects are passed to action objects as parameters when coala tries to apply an action. This is a bit restrictive as this means that coala needs to have a global set of actions that it tries to apply on all `Result` objects. Sure, coala checks if an action `is_applicable()` before it tries to apply it but it needs to do that for every `Result` that every bear ever generates.
+
+We need a way allow results to define actions by themselves, like result = Result(actions=[AmendCommitAction, ...]). Here are the benefits of such an approach:
+1. More control inside actions over the interaction process.
```

Comment:
That's not the benefit of passing actions with results, this is something additional we should do so actions can do more.